### ISSUE TITLE
Javascript mode - Undeclared/not initialized variables don't throw error.

### ISSUE THREAD

**Issue by madhavan020985** [flycheck/flycheck#588]
In this snippet, i would like to see an error at line number 4 where undeclared variable 'asdadfas' is used. But flycheck is not throwing such an error....
````javascript
var a = require('b')
var c = function(b,c){
  console.log("..............")
  console.log(asdadfas)
}
````

**Comment by lunaryorn** [comment]
@madhavan020985 Is Flycheck Mode even active in the buffer?  If yes, what syntax checker are you using?

**Comment by kimgronqvist** [comment]
Are you using jshint? If so, this is not turned on by default. You need to put /*jshint undef:true*/ at the top of the file or include it in a .jshintrc.

**Comment by madhavan020985** [comment]
hi kimgron, now undefined objects are highlighted. But in the same snippet above, globally declared objects/functions as require & console are also highlighted as undefined....
@lunaryorn, i din't specify syntax checker from my side. Major mode is javascript-mode & minor mode is flycheck-mode.. so, yes flycheck is active in the buffer.

**Comment by kimgronqvist** [comment]
You can whitelist certain globals with the "globals" options: http://jshint.com/docs/options/#globals

    /*jshint undef:true*/
    /*global console: true, require: true */

This is not Flycheck related.

**Comment by madhavan020985** [comment]
done, it works... i will close this issue
